### ISSUE TITLE
Ingestion fails on 'Connection reset by peer' when using native parallel ingestion

### ISSUE THREAD

**Issue by dark-spark2** [apache/druid#9467]
### Affected Version
druid 0.17
### Description
When running native ingestion with "index_parallel" on druid 0.17. 
I receive the following error:
`java.io.IOException: java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.io.IOException: Connection reset by peer 
at org.apache.druid.java.util.common.FileUtils.copyLarge(FileUtils.java:311) ~[druid-core-0.17.0.jar:0.17.0]
	at org.apache.druid.indexing.common.task.batch.parallel.PartialSegmentMergeTask.fetchSegmentFile(PartialSegmentMergeTask.java:261) ~[druid-indexing-service-0.17.0.jar:0.17.0]
	at org.apache.druid.indexing.common.task.batch.parallel.PartialSegmentMergeTask.fetchSegmentFiles(PartialSegmentMergeTask.java:236) ~[druid-indexing-service-0.17.0.jar:0.17.0]
	at org.apache.druid.indexing.common.task.batch.parallel.PartialSegmentMergeTask.runTask(PartialSegmentMergeTask.java:178) ~[druid-indexing-service-0.17.0.jar:0.17.0]
	at org.apache.druid.indexing.common.task.batch.parallel.PartialGenericSegmentMergeTask.runTask(PartialGenericSegmentMergeTask.java:44) ~[druid-indexing-service-0.17.0.jar:0.17.0]
	at org.apache.druid.indexing.common.task.AbstractBatchIndexTask.run(AbstractBatchIndexTask.java:138) ~[druid-indexing-service-0.17.0.jar:0.17.0]
	at org.apache.druid.indexing.overlord.SingleTaskBackgroundRunner$SingleTaskBackgroundRunnerCallable.call(SingleTaskBackgroundRunner.java:419) [druid-indexing-service-0.17.0.jar:0.17.0]
	at org.apache.druid.indexing.overlord.SingleTaskBackgroundRunner$SingleTaskBackgroundRunnerCallable.call(SingleTaskBackgroundRunner.java:391) [druid-indexing-service-0.17.0.jar:0.17.0]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]`

I am sending the following task to overlord:
`{
  "type" : "index_parallel",
  "spec" : {
    "ioConfig" : {
      "type": "index_parallel",
      "inputSource": {
        "type": "s3",
        "prefixes": ["s3://filePAth"]
      },
      "inputFormat": {
        "type": "json"
      }
    },
    "dataSchema" : {
      "dataSource" : "test_batch",
      "granularitySpec" : {
        "type" : "uniform",
        "segmentGranularity" : "HOUR",
        "queryGranularity" : "HOUR",
        "intervals" : ["2018-10-06/2018-10-07"]
      },
      "dimensionsSpec" : {
        "dimensions" : [
          "filed1","field2","field3"
        ]
        },
        "timestampSpec" : {
            "format" : "auto",
            "column" : "timestamp"
        },
    },
    "tuningConfig" : {
      "type": "index_parallel",
      "maxNumConcurrentSubTasks": 2,
      "partitionsSpec" : {
       "type": "single_dim",
       "targetRowsPerSegment": 100000,
       "partitionDimension": "field1"
      }
    }
  }
}

`
The strange thing is that when removing the "partitionSpec" everything works well.
Thanks.

**Comment by liran-funaro** [comment]
Could you post here the logs of all your indexer tasks?

**Comment by morrifeldman** [comment]
I think I ran into the same failure when trying single_dim indexing.  The stack trace in my case was:

```
2020-06-17T22:22:58,075 ERROR [task-runner-0-priority-0] org.apache.druid.indexing.overlord.SingleTaskBackgroundRunner - Exception while r
unning task[AbstractTask{id='partial_dimension_distribution_ltv_jnmjigoc_2020-06-17T22:22:29.979Z', groupId='index_parallel_ltv_bppfkkka_2
020-06-17T22:22:24.488Z', taskResource=TaskResource{availabilityGroup='partial_dimension_distribution_ltv_jnmjigoc_2020-06-17T22:22:29.979
Z', requiredCapacity=1}, dataSource='ltv', context={forceTimeChunkLock=false}}]
java.lang.RuntimeException: java.io.IOException: No such file or directory
        at org.apache.druid.data.input.impl.InputEntityIteratingReader.lambda$read$0(InputEntityIteratingReader.java:81) ~[druid-core-0.18
.1-hadoop-2.9.2-af.jar:0.18.1-hadoop-2.9.2-af]
...
some lines omitted here
...
2020-06-17T22:22:58,079 INFO [task-runner-0-priority-0] org.apache.druid.indexing.worker.executor.ExecutorLifecycle - Task completed with status: {
  "id" : "partial_dimension_distribution_ltv_jnmjigoc_2020-06-17T22:22:29.979Z",
  "status" : "FAILED",
  "duration" : 100,
  "errorMsg" : "java.lang.RuntimeException: java.io.IOException: No such file or directory",
  "location" : {
    "host" : null,
    "port" : -1,
    "tlsPort" : -1
  }
}
```

This issue is fixed in #10046 by creating the missing temporary directory.
